vespalib/src/vespa/vespalib/datastore/datastorebase.cpp

```diff
@@ -164,6 +157,23 @@ DataStoreBase::consider_grow_active_buffer(uint32_t type_id, size_t elems_needed
     return true;
 }
 
+uint32_t
+DataStoreBase::getFirstFreeBufferId() {
+    for (uint32_t buffer_id = 0; buffer_id < _states.size(); buffer_id++) {
+        if (getBufferState(buffer_id).isFree()) {
+            return buffer_id;
+        }
+    }
+    // Need next(new) buffer
+    return _states.size();
```

Comment:
DataStoreBase::onActive() needs to assert that buffer id is reasonable to avoid a buffer overflow.

Reply:
Correct, that is part of upcomming PR.